environments/common/src/main/java/org/jboss/weld/environment/deployment/discovery/ClassPathBeanArchiveScanner.java

```diff
@@ -114,11 +126,20 @@ private void scanDirectory(File entryDirectory, ImmutableList.Builder<ScanResult
                 results.add(new ScanResult(null, entryDirectory.getPath()));
             }
         }
+
+        File manifestFile = new File(entryDirectory, MANIFEST_FILE);
+        if (manifestFile.canRead()) {
+            final Manifest manifest = new Manifest(new FileInputStream(manifestFile));
+            final Attributes manifestMainAttributes = manifest.getMainAttributes();
+            if (manifestMainAttributes.containsKey(CLASS_PATH)) {
+                scanManifestClassPath(entryDirectory.toURI().toURL(), manifestMainAttributes.getValue(CLASS_PATH), results);
+            }
+        }
     }
 
     private void scanJarFile(File entryFile, ImmutableList.Builder<ScanResult> results) throws IOException {
-        try (ZipFile zip = new ZipFile(entryFile)) {
-            ZipEntry beansXmlEntry = zip.getEntry(AbstractWeldDeployment.BEANS_XML);
+        try (JarFile jar = new JarFile(entryFile)) {
+            ZipEntry beansXmlEntry = jar.getEntry(AbstractWeldDeployment.BEANS_XML);
```

Comment:
Since you are now using `JarFile`, `ZipEntry` can be changed to `JarEntry` I suppose.

Reply:
No, unless I use `getJarEntry()` instead of `getEntry()`.
Which I didn't do as I don't use anything specific to `JarEntry`.
I only changed from `ZipFile` to `JarFile` to easily get hold of the manifest via the API instead of searching the `ZipEntry` by path.
If you want me to change it for consitency reasons, tell me, but I don't think this is necessary.

Reply:
Yes, I was thinking of consistency reasons here.
It's probably nitpicking, but I would prefer to change it :-)

Reply:
done